### ISSUE TITLE
Scaling exactGP regression with KissGP fails

### ISSUE THREAD

**Issue by marcglettig** [cornellius-gp/gpytorch#1124]
I have implemented my problem with ExactGP regression and the model works perfectly fine. Now I am interested in increasing my number of data points. I therefore tried to implement a GPRegressionModel with a GridInterpolationKernel. But the training turns out to be much slower as well as the score to be extremely bad. After some consideration I assume my KISS model might be converging to a local minimum, while this does not explain the much longer training time. I assume that I might have fundamentally misunderstood how to apply the KISSGP method but I do not see where.

My model has 3 (to be scaled up as well) dimensional X at the moment. ExactGP works fine although training obviously happens to be slow with more than 2000 data points, but I would like to scale up to 25000.

### To reproduce

Working exact GP regression model:
```python
    class ExactGPModel(gpytorch.models.ExactGP):
        def __init__(self, train_x, train_y, likelihood):
            super(ExactGPModel, self).__init__(train_x, train_y, likelihood)
            self.mean_module = gpytorch.means.ConstantMean()
            self.covar_module = gpytorch.kernels.ScaleKernel(gpytorch.kernels.RBFKernel())
            
        def forward(self, x):
            mean_x = self.mean_module(x)
            covar_x = self.covar_module(x)
            return gpytorch.distributions.MultivariateNormal(mean_x, covar_x)

    likelihood = gpytorch.likelihoods.GaussianLikelihood()
    model = ExactGPModel(train_x, train_y, likelihood)
```

Slow KISS-GP model:
```python
    class GPRegressionModel(gpytorch.models.ExactGP):
        def __init__(self, train_x, train_y, likelihood):
            super(GPRegressionModel, self).__init__(train_x, train_y, likelihood)

            grid_size = gpytorch.utils.grid.choose_grid_size(train_x, 1.0)

            self.mean_module = gpytorch.means.ConstantMean()
            self.covar_module = gpytorch.kernels.ScaleKernel(
                gpytorch.kernels.GridInterpolationKernel(
                    gpytorch.kernels.RBFKernel(), grid_size=int(grid_size), num_dims=train_x.shape[1]))

        def forward(self, x):
            mean_x = self.mean_module(x)
            covar_x = self.covar_module(x)
            return gpytorch.distributions.MultivariateNormal(mean_x, covar_x)

    likelihood = gpytorch.likelihoods.GaussianLikelihood()
    model = GPRegressionModel(train_x, train_y, likelihood)
```

### Expected Behavior
I would assume the KISS-GP model to train much faster as we are exploiting the grid structure. Although I would expect a small loss in performance of the model, I do not understand the convergence to almost a constant. 

So this basically boils down to two questions:
- How can the training be this much slower with KISS-GP?
- How can I avoid converging to a local minimum with KISS-GP (changing the length scale initialization to a smaller value did not help)?

**Comment by jacobrgardner** [comment]
With 2000 data points (and quite possibly even 25000 at this point), it's not particularly surprising that KISS-GP would be slower than exact GPs. All of the approximate methods will be slower with that little data. Anything that does operations other than simple matmuls is going to be slower on actual GPUs until asymptotic complexity actually starts winning.

In terms of model performance, we'd really have to see a fully working example to speculate. What grid_size is the `choose_grid_size` recommending for your data?

**Comment by Thunderabc** [comment]
I have encountered a similar issue. I'm working on a three-dimensional application with datapoints = 50000, and the automatically selected grid_size = 36. The process runs slowly and often prompts "Kernel died," preventing further iterations.
